position_sender_node:
  ros__parameters:
    real_position_topic: real_position
    short_motor_srv: short_motor
    update_position_topic: update_position
    update_period: 0.1

orchestrator_gui_node:
  ros__parameters:
    set_asserv: false
    position_topic: position_out
    command_topic: command
    feedback_command_topic: feedback_command

tf_broadcaster_node:
  ros__parameters:
    update_position_topic: update_position
    visualization_topic: visualization
    init_visu_srv: init_visu
    default_color: yellow
    available_colors: ["yellow", "blue"]
    boundaries: [0.0, 3.0, 0.0, 2.0]
    beacons: [-0.094, 0.072, -0.094, 1.928, 3.094, 1., 1.275, 2.12]

main_robot:
  beacon_detector_node:
    ros__parameters:
      enable_robot_position_reception: False
      object_topic: raw_obstacles
      position_topic: position_out
      debug_topic: debug
      color_topic: init/team_color
      default_color: yellow
      available_colors: ["yellow", "blue"]
      beacons: [-0.094, 0.072, -0.094, 1.928, 3.094, 1., 1.275, 2.12]
      boundaries:  [0.0, 3.0, 0.0, 2.0]
      init_position: [1.77, 0.23, 0.0]
      distance_tolerance: 0.3
      angle_tolerance: 0.7853981633974483
      robot_position_topic: robot_position_input
      enable_wait_color: False
      enable_initial_position: False

  lidar_simulation_node:
    ros__parameters:
      object_topic: raw_obstacles
      scan_topic: scan
      update_position_topic: update_position
      color_service: init/team_color
      default_color: yellow
      available_colors: ["yellow", "blue"]
      beacons: [-0.094, 0.072, -0.094, 1.928, 3.094, 1., 1.275, 2.12]
      boundaries:  [0.0, 3.0, 0.0, 2.0]

      use_lidar_points: true
      lidar_range: 10.
      radius: 0.05
      num_points: 600
      angle_info: [0., 360.]

  nav_simulation_node:
    ros__parameters:
      moveto_action: moveto
      real_position_topic: real_position
      trigger_position_srv: trigger_position
      random_moves: False
      boundaries:  [0.0, 3.0, 0.0, 2.0]
      compute_period: 0.01
      angular_velocity: 0.4
      linear_velocity: 0.15

  obstacle_extractor_node:
    ros__parameters:
      active: True
      use_scan: True
      use_pcl: False
      use_sim_time: True

      use_split_and_merge: True
      circles_from_visibles: True
      discard_converted_segments: True
      transform_coordinates: True

      min_group_points: 2

      max_group_distance: 0.12
      distance_proportion: 0.00628
      max_split_distance: 0.2
      max_merge_separation: 0.2
      max_merge_spread: 0.2
      max_circle_radius: 0.2
      radius_enlargement: 0.05
      max_x_range: 4.0
      max_y_range: 4.0

      frame_id: laser_frame

ennemi_robot:
  beacon_detector_node:
    ros__parameters:
      enable_robot_position_reception: False
      object_topic: raw_obstacles
      position_topic: position_out
      debug_topic: debug
      color_topic: init/team_color
      default_color: yellow
      available_colors: ["yellow", "blue"]
      beacons: [-0.094, 0.072, -0.094, 1.928, 3.094, 1., 1.275, 2.12]
      boundaries:  [0.0, 3.0, 0.0, 2.0]
      init_position: [1.77, 0.23, 0.0]
      distance_tolerance: 0.3
      angle_tolerance: 0.7853981633974483
      robot_position_topic: robot_position_input
      enable_wait_color: False
      enable_initial_position: False

  lidar_simulation_node:
    ros__parameters:
      object_topic: raw_obstacles
      scan_topic: scan
      update_position_topic: update_position
      color_service: init/team_color
      default_color: yellow
      available_colors: ["yellow", "blue"]
      beacons: [-0.094, 0.072, -0.094, 1.928, 3.094, 1., 1.275, 2.12]
      boundaries:  [0.0, 3.0, 0.0, 2.0]

      use_lidar_points: false
      lidar_range: 10.
      radius: 0.05
      num_points: 600
      angle_info: [0., 360.]

  nav_simulation_node:
    ros__parameters:
      moveto_action: moveto
      real_position_topic: real_position
      trigger_position_srv: trigger_position
      random_moves: False
      boundaries:  [0.0, 3.0, 0.0, 2.0]
      compute_period: 0.01
      angular_velocity: 0.4
      linear_velocity: 0.15

  obstacle_extractor_node:
    ros__parameters:
      active: True
      use_scan: True
      use_pcl: False
      use_sim_time: True

      use_split_and_merge: True
      circles_from_visibles: True
      discard_converted_segments: True
      transform_coordinates: True

      min_group_points: 2

      max_group_distance: 0.12
      distance_proportion: 0.00628
      max_split_distance: 0.2
      max_merge_separation: 0.2
      max_merge_spread: 0.2
      max_circle_radius: 0.3
      radius_enlargement: 0.2

      frame_id: laser_frame
